File: 946-Validate Stack Sequences.py
class Solution(object):
    def validateStackSequences(self, pushed, popped):
        """
        :type pushed: List[int]
        :type popped: List[int]
        :rtype: bool
        """
        j = 0
        stack = []
        
        for i in range(len(pushed)):
            stack.append(pushed[i])
            while stack and stack[-1] == popped[j]:
                stack.pop()
                j+=1
        
        return j == len(popped)

File: C++/946-Validate Stack Sequences.cpp
class Solution {
public:
    bool validateStackSequences(vector<int>& pushed, vector<int>& popped) {
        stack<int> stk;
        int ind = 0;
        for (int c:pushed){
            stk.push(c);
            while (!stk.empty() && stk.top() == popped[ind]){
                ind += 1;
                stk.pop();
            }
        }
        return stk.empty();
    }
};